internal/core/metadata/operators/addressable/delete.go

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package addressable
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/errors"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type DeleteExecutor interface {
+	Execute() error
+}
+
+type addressDelete struct {
+	database   AddressDeleter
+	identifier string
+}
+
+// This method adds the provided Addressable to the database.
+func (op addressDelete) Execute() error {
+	// Check if the addressable exists
+	idAddr, idErr := op.database.GetAddressableById(op.identifier)
+	nameAddr, nameErr := op.database.GetAddressableByName(op.identifier)
+
+	// if this case hits, then it's safe to say we don't have a usable addressable
+	if idErr != nil && nameErr != nil {
+		if idErr == db.ErrNotFound && nameErr == db.ErrNotFound {
```

Comment:
This check isn't working as intended and the function exits on line 46 below. This results in a failing blackbox test.
<img width="1073" alt="Screen Shot 2019-08-05 at 1 19 22 PM" src="https://user-images.githubusercontent.com/1342980/62486138-0fca6800-b784-11e9-8fee-37a16928953c.png">